circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/Replica.java

```diff
@@ -98,14 +98,21 @@ public void updateMetadata(
     try (CloseableMetaStoreClient client = getMetaStoreClientSupplier().get()) {
       Optional<Table> oldReplicaTable = updateTableMetadata(client, eventId, sourceTable, replicaDatabaseName,
           replicaTableName, locationManager.getTableLocation(), replicationMode);
-      if (oldReplicaTable.isPresent() && LocationUtils.hasLocation(oldReplicaTable.get())) {
+      if (oldReplicaTable.isPresent()
+          && LocationUtils.hasLocation(oldReplicaTable.get())
+          && isUnpartitioned(oldReplicaTable.get())) {
```

Comment:
I opted for this fix so we'll never add the location of an unpartitioned table.

Alternatively we could change the call, to call the updateMetadata with the partitions argument and make sure that works ok if partitions is an empty list. I've opted for above fix as I think any accidental call to this method should never schedule the table location for a partitioned table.